src/execution/lineage/operator_lineage.cpp

```diff
@@ -8,11 +8,15 @@
 namespace duckdb {
 
 void OperatorLineage::Capture(const shared_ptr<LineageData>& datum, idx_t lineage_idx) {
-	// Prepare this vector's chunk to be passed on to future operators
-	pipeline_lineage->AdjustChunkOffsets(datum->Count(), lineage_idx);
-
-	// Capture this vector
-	idx_t offset = pipeline_lineage->GetChildChunkOffset(lineage_idx);
+	if (!trace_lineage) return;
+	idx_t offset = 0;
+	if (pipeline_lineage) {
+		// Prepare this vector's chunk to be passed on to future operators
+		pipeline_lineage->AdjustChunkOffsets(datum->Count(), lineage_idx);
+
+		// Capture this vector
+		offset = pipeline_lineage->GetChildChunkOffset(lineage_idx);
+	}
```

Comment:
Which one is this for? DelimJoin?

Reply:
yes. also generally after the change you pushed for group bys isn't it the case the this adjustment isn't necessary?